systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.operators;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.KafkaTopic;
+import io.strimzi.operator.common.Annotations;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.resources.ResourceOperation;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.crd.kafkaclients.KafkaBasicExampleClients;
+import io.strimzi.systemtest.resources.operator.BundleResource;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaTopicTemplates;
+import io.strimzi.systemtest.utils.ClientUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaTopicUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.JobUtils;
+import io.strimzi.systemtest.utils.kubeUtils.controllers.StatefulSetUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import io.strimzi.test.annotations.IsolatedTest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Feature Gates should give us additional options on
+ * how to control and mature different behaviors in the operators.
+ * https://github.com/strimzi/proposals/blob/main/022-feature-gates.md
+ */
+@Tag(REGRESSION)
+public class FeatureGatesST extends AbstractST {
+
+    static final String NAMESPACE = "feature-gates-tests";
+    private static final Logger LOGGER = LogManager.getLogger(FeatureGatesST.class);
+
+    protected void deployClusterOperatorWithEnvVars(ExtensionContext extensionContext, List<EnvVar> envVars) {
+        if (Environment.isOlmInstall() || Environment.isHelmInstall()) {
+            installClusterOperator(extensionContext, NAMESPACE);
+        } else {
+            prepareEnvForOperator(extensionContext, NAMESPACE, Arrays.asList(NAMESPACE, NAMESPACE));
+            applyBindings(extensionContext, NAMESPACE);
+            resourceManager.createResource(extensionContext, BundleResource.clusterOperator(NAMESPACE, envVars).build());
```

Comment:
What if I will not set env var for feature gates and will try to install CO via OLM?

Reply:
Then you are running this test incorrectly. It has been stated multiple times to me, that only Bundle installation type should be used w/o env var. :-) 
OLM and Helm should use external Var set (optionally for Bundle).

Reply:
so what's the purpose of this test for HELM and OLM as you have it now? It kinda doesn't make sense because it will test the same thing which is already tested by other tests. You should have there only bundle install type and assumption, that `CLUSTER_OPERATOR_INSTALL_TYPE=bundle`

Reply:
Removed running this testcase with OLM and Helm installations after offline talk. -> `assumeFalse()`